weed/s3api/s3api_bucket_handlers.go

```diff
@@ -135,6 +135,13 @@ func (s3a *S3ApiServer) DeleteBucketHandler(w http.ResponseWriter, r *http.Reque
 	}
 
 	err := s3a.WithFilerClient(func(client filer_pb.SeaweedFilerClient) error {
+		isEmpty, err := s3a.isDirectoryAllEmpty(client, s3a.option.BucketsPath, bucket)
+		if err != nil {
+			return fmt.Errorf("check empty bucket %s: %v", bucket, err)
+		}
+		if !isEmpty {
+			return fmt.Errorf("BucketNotEmpty")
```

Comment:
With this PR, need to iterate and recursively delete all entries first, before deleting the bucket. This is a slow process if the bucket has a lot of files, not what most users want.

Reply:
> With this PR, need to iterate and recursively delete all entries first, before deleting the bucket. This is a slow process if the bucket has a lot of files, not what most users want.

For the sake of safety and to prevent accidental deletion of a bucket with all its contents, I propose to remove this function as an option or only admin of all buckets to delete recursively ?